src/physics/base/CutoffParams.hh

```diff
@@ -0,0 +1,84 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file CutoffParams.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "CutoffInterface.hh"
+#include "base/CollectionMirror.hh"
+#include "physics/base/Units.hh"
+#include "physics/base/ParticleParams.hh"
+#include "physics/material/MaterialParams.hh"
+
+#include <vector>
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Data management for particle and material cutoff values.
+ *
+ * Geant4 provides accessors to its production cuts from its
+ * \c G4MaterialCutsCouple class, which couples cutoff and material data.
+ * During import, for simplicity, G4's production cuts are stored alongside
+ * the material information, in \c ImportMaterial . Since this is a direct
+ * import, the cutoff map in \c ImportMaterial stores only the cuts available
+ * in Geant4, i.e. only values for gammas, electrons, positrons, and protons.
+ *
+ * In Celeritas, particle cutoff is stored contiguously in a single vector
+ * of size num_particles * num_materials, which stores all particle cutoffs
+ * for all materials. During import, any particle that is not in Geant4's
+ * list receives a zero cutoff value. This opens the possibility to expand
+ * cutoffs in the future, when data is not imported anymore.
```

Comment:
From our discussion, I'd intended for the G4Importer *not* to have to to construct cutoffs for unused particles --  this will eliminate the need to sort the particle cutoff types in the exporter, and there's also the minor plus of reducing storage requirements.

Maybe to make this clearer I should've suggested the cutoff inputs be a `std::map<PDGNumber, vector<Cutoff>>`?

Reply:
Just to be sure I understood the update: the `geant-exporter` does not store unused particles in the root file (`ImportMaterial` already stores said `map<pdg, vector<cutoffs>>`. On the other hand, `CutoffParams` stores values for all particles, so we could expand our cutoff criteria in the future. So you prefer to store in `CutoffParams` only the 4 particle cutoffs per material, and require the `ParticleId` passed at construction time to `CutoffView` to match one of the 4 valid types (electron, positron, gamma, or proton)?

Reply:
Oh ok, I guess I didn't look closely enough at the exporting data since we're going to overhaul the MaterialMap stuff soon. So the exporter is good to go then, *but* I think essentially the remapping of PDG number to particle ID should be done inside the CutoffParams constructor, not inside the RootImporter.  So a unit test (for example) would be able to specify all the particles or just a single particle, and not have to bother putting in empty ones for unused particles.

And then The `RootImporter` should reinstate the sorting of the `ParticleParams`. Basically the expansion of sparse particle identifiers (PDG) to dense particle IDs should be done in the CutoffParams rather than the RootImporter.

Make sense? Sorry for the confusion. 😕